### PR TITLE
Add type filter for activities log page

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/5850

⚠️ Require: https://github.com/opencollective/opencollective-api/pull/7846

https://user-images.githubusercontent.com/12435965/185206651-3ee21111-611d-4d6b-9975-61175db3f4cc.mp4

### CODE REVIEW

File: components/admin-panel/sections/ActivityLog/ActivityTypeFilter.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  const ActivityTypeKeys = Object.keys(ActivityClasses);
+  ActivityTypeKeys.unshift('ALL');
+
+  return (
+    <StyledSelectFilter
+      inputId="activity-type-filter"
+      onChange={({ value }) => onChange(value)}
+      value={getOption(value || 'ALL')}
+      options={ActivityTypeKeys.map(getOption)}
```
Comment (Betree) [inline components/admin-panel/sections/ActivityLog/ActivityTypeFilter.js:22]: As mentioned by PM, we're not interested in all activity classes: let's filter out reports

File: lib/i18n/activities.js [hunk 2]
```diff
@@ -49,3 +49,38 @@ export const ActivityLabelI18n = defineMessages({
...
+    defaultMessage: 'Virtual Card activity',
+  },
+  FUND_EVENTS: {
+    id: 'FundEventActivity',
+    defaultMessage: 'Fund Event activity',
+  },
+  REPORTS: {
+    id: 'ReportsActivity',
+    defaultMessage: 'Reports activity',
+  },
```
Comment (Betree) [inline lib/i18n/activities.js:85]: Sounds like all these should be plural (e.g. "Expenses activities"). Worth sharing a screenshot on Slack's #wording channel to ask for some feedback.

File: lib/i18n/activities.js [hunk 3]
```diff
@@ -49,3 +49,38 @@ export const ActivityLabelI18n = defineMessages({
...
+    id: 'UpdatesActivity',
+    defaultMessage: 'Updates activity',
+  },
+  VIRTUAL_CARDS: {
+    id: 'VirtualCardActivity',
+    defaultMessage: 'Virtual Card activity',
+  },
+  FUND_EVENTS: {
+    id: 'FundEventActivity',
+    defaultMessage: 'Fund Event activity',
```
Comment (Betree) [inline lib/i18n/activities.js:80]:
```suggestion
    defaultMessage: 'Fund & Event activities',
```

**Review by Betree — COMMENTED** [review]
Great first version for this feature, let's polish it a bit

File: components/admin-panel/sections/ActivityLog/ActivityTypeFilter.js (hunk 1 above)
Comment (SudharakaP) [reply components/admin-panel/sections/ActivityLog/ActivityTypeFilter.js:22]: Thanks, I've filtered out reports. 👍🏽

File: lib/i18n/activities.js (hunk 2 above)
Comment (SudharakaP) [reply lib/i18n/activities.js]: Changed everything to "... activities" and created a thread on the wording channel; https://opencollective.slack.com/archives/C019YCG6RCY/p1660831885324169

File: lib/constants/activityTypes.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+export default {
```
Comment (Betree) [inline lib/constants/activityTypes.js:1]: Is that file used somewhere?

**Review by Betree — COMMENTED** [review]
Let's wait for some feedback on the labels, the code itself looks good :+1:

File: lib/constants/activityTypes.js (hunk 4 above)
Comment (SudharakaP) [reply lib/constants/activityTypes.js]: Ops, this is a leftover from my previous commit. I've removed this file. 👍🏽

**Comment by SudharakaP** [comment]
> Let's wait for some feedback on the labels, the code itself looks good +1

@Betree : Based on the feedback at https://opencollective.slack.com/archives/C019YCG6RCY/p1660831885324169, I've changed the labels to remove the word `activity`.

image

**Review by Betree — APPROVED** [review]
Perfect, and the fact that we're re-using existing i18n strings will simplify the lives of translators. :100:

**Pull request merged**